src/conv-core/convcore.C

```diff
@@ -2510,19 +2514,19 @@ static CmiReduction* CmiGetReductionCreate(int id, short int numChildren) {
 }
 
 static void CmiClearReduction(int id) {
-  const int idx = id & ~((~0u) << CmiLogMaxReductions);
-  auto & redref = CpvAccess(_reduce_info)[idx];
+  auto & redref = CpvAccess(_reduce_info)[CmiGetRedIndex(id)];
   auto red = redref;
   redref = nullptr;
   free(red);
 }
 
-static CmiReduction* CmiGetNextReduction(short int numChildren) {
-  int id = CpvAccess(_reduce_seqID_global);
-  int newid = id + CmiReductionID_multiplier;
-  if (id > 0xFFF0) newid = CmiReductionID_globalOffset;
+static CmiReductionID CmiGetNextReductionID() {
```

Comment:
Minor nitpick, file convention seems to be `(void)` vs. `()`.

Reply:
That's a holdover from when this file was `convcore.c`. I'm not eager to add the `void` because the file is C++ and the function has no declaration in a header that could be included from C. There are other inconsistencies in this file too, like whether pointers and references have a space to the right of their respective symbols.

Reply:
I'll fix it anyway since I'm having to make more changes to this PR.